extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js

```diff
@@ -35,44 +39,101 @@ promClient.register.setDefaultLabels({
  * @property {string} [serviceName] - Name of our service to match generic metrics
  */
 
-const replicationStatusMetric = new promClient.Counter({
-    name: 'replication_status_changed_total',
-    help: 'Number of objects updated',
-    labelNames: ['origin', 'containerName', 'replicationStatus'],
-});
-
-const kafkaLagMetric = new promClient.Gauge({
-    name: 'kafka_lag',
-    help: 'Number of update entries waiting to be consumed from the Kafka topic',
-    labelNames: ['origin', 'containerName', 'partition', 'serviceName'],
-});
-
-const replayAttempts = new promClient.Counter({
-    name: 'replication_replay_attempts_total',
-    help: 'Number of total attempts made to replay replication',
-    labelNames: ['origin', 'containerName'],
-});
-
-const replaySuccess = new promClient.Counter({
-    name: 'replication_replay_success_total',
-    help: 'Number of times an object was replicated during a replay',
-    labelNames: ['origin', 'containerName'],
-});
-
 /**
  * Contains methods to incrememt different metrics
  * @typedef {Object} ReplicationStatusMetricsHandler
  * @property {CounterInc} status - Increments the replication status metric
  * @property {GaugeSet} lag - Set the kafka lag metric
  * @property {CounterInc} replayAttempts - Increments the replay attempts metric
  * @property {CounterInc} replaySuccess - Increments the replay success metric
+ * @property {CounterInc} replayQueuedObjects - Increments the replay queued objects metric
+ * @property {CounterInc} replayQueuedBytes - Increments the replay queued bytes metric
+ * @property {CounterInc} replayQueuedFileSizes - Increments the replay queued file sizes metric
+ * @property {CounterInc} replayCompletedObjects - Increments the replay completed objects metric
+ * @property {CounterInc} replayCompletedBytes - Increments the replay completed bytes metric
+ * @property {CounterInc} replayCompletedFileSizes - Increments the replay completed file sizes metric
+ */
+
+/**
+ * @param {Object} repConfig - Replication configuration
+ * @returns {ReplicationStatusMetricsHandler} Metric handlers
  */
-const metricsHandler = {
-    status: wrapCounterInc(replicationStatusMetric),
-    lag: wrapGaugeSet(kafkaLagMetric),
-    replayAttempts: wrapCounterInc(replayAttempts),
-    replaySuccess: wrapCounterInc(replaySuccess),
-};
+function loadMetricHandlers(repConfig) {
```

Comment:
Metrics need to be loaded post constructor since we need some configs

Reply:
This file is pretty large, would it be a good idea to move the metrics logic to its own file?

Reply:
Generally I would say yes, but with having to keep forward ports relatively easy it might be easier to keep it here even though it is quite large. I am a little concerned on switching to a function instead of top level consts in the general scheme. I do like this pattern as it allows a lot more configurations and testing. But they are still tied to the global prometheus registry, which is not ideal. Maybe we could find a pattern that fits our work and adapt it moving forward.